### DISCUSSION TITLE
Connect to different Postgres Servers

### DISCUSSION THREAD

**Discussion by pc-dok** [dalibo/ldap2pg#629] [Q&A]
Hi all

I am wondering if i am the only one, but for me it is not clear how i can use ldap2pg on a central server in our environment, where i have let say, for every postgres server a yaml config. so that i can manage all postgres server from this server. so i struggeling in the creation of the yaml config, to find out, where i must add my servers, so that ldap2pg know, where he must go. LDAP is clear for me, but where i define the remote postgres servers? is it something with DSN or so. Is there a example how to connect to remote postgres servers? sorry for asking, but in the example, i dont find this point!

regards
franco

**Comment by bersace** [comment]
Hi @pc-dok , thanks for reaching !

This depends on the variation between configurations.
I suggest to have a directory per server with a `.env` file containing PG* en vars. something like:

``` console
$ tree
.
├── ldaprc
├── server1/
│   ├── .env
│   └── ldap2pg.yml
├── server2/
│   ├── .env
│   └── ldap2pg.yml
.
.
.
```

ldap2pg 6.1 (unreleased) will load `.env` autoamatically. For new, use a bash wrapper to implement this architecture.

**Comment by pc-dok** [comment]
hi etienne

ok, is a bit more clear now, but what looks like a .env file than? have you a short example? maybe i write you my env.

Postgres Server 16.2
Server: testpg.example.com
IP: 10.10.10.1
Instance: pgitest01
Port: 61101

Sorry for that many questions, but i am absolut newbie in this ldap2pg and my thinking is, that i can handle all my postgres servers than from one central mgmgt server!

regards
franco

**Reply by bersace** [reply]
You must run ldap2pg for each server. You can't tell ldap2pg to synchronize all servers at once.

See .env is an old Heroku de factor standard. For you case, it's simply a shell script to be sourced.

**Comment by pc-dok** [comment]
something like this you mean?

POSTGRES_USER="someuser"
POSTGRES_PASSWORD="somepwd"
POSTGRES_DB="somedb"
POSTGRES_URL="postgres://${POSTGRES_USR}:${POSTGRES_PWD}@postgres:5432/${POSTGRES_DB}?sslmode=disable"

**Comment by pc-dok** [comment]
ok, i found out now how i can connect to the other servers over export from the correct PG* ENV Variables. I only now get this error:

15:13:30 DEBUG  Closing Postgres global connection.              database=test
15:13:30 ERROR  Fatal error.                                     err="LDAP Result Code 206 \"Empty password not allowed by the client\": ldap: empty password not allowed by the client"

what does this mean?

regards
franco

**Reply by bersace** [reply]
Do you need anonymous auth against your directory ?
